src/classes/ADDR_Addresses_TEST.cls

```diff
@@ -0,0 +1,562 @@
+        System.assertEquals(System.today(), cityByAddress.get('Boston').Latest_Start_Date__c);
+        
+    }
+    
+    /**********************************************************************************************************
+    * @description Test method that specifically verifies the following for beforeUpdate:
+    * (1) When an existing Address record is updated and it doesn't match the filter, and Default_Address__c is set
+    * to false, it should populate Latest_End_Date__c with today's date. 
+    * (2) When an existing Address record is updated and it doesn't match the filter, and Default_Address__c is set 
+    * to true, it should populate Latest_Start_Date__c with today's date. 
+    * (3) When an existing Address record is updated and it does match the filter, and Default_Address__c is set 
+    * to false, it should not populate Latest_End_Date__c with today's date. 
+    * (4) When an existing Address record is updated and it doesn't match the filter, and Default_Address__c is set
+    * to true, it should not populate Latest_Start_Date__c with today's date. 
+    */
+       @isTest
+    private static void beforeUpdateAddress() {
+        UTIL_CustomSettingsFacade.getSettingsForTests(new Hierarchy_Settings__c(Account_Processor__c = UTIL_Describe_API.getAdminAccRecTypeID(),
+                                                                               Contacts_Addresses_Enabled__c = true));
+        
+        //Retrieves the defaultTDTMConfig 
+        List<TDTM_Global_API.TdtmToken> tokens = TDTM_Global_API.getDefaultTdtmConfig(); 
+        
+        UTIL_UnitTestData_TEST.ContactsWithAddrs contactsAddrs = UTIL_UnitTestData_TEST.createTestContactsAddrs(2);
+        List<Contact> contacts = contactsAddrs.contacts; 
+        List<Address__c> addresses = contactsAddrs.addrs;
+        
+        //Add filter on ADDR_Addresses_TDTM triggerHandler
+        for(TDTM_Global_API.TdtmToken token : tokens) {
+            if(token.className == 'ADDR_Addresses_TDTM') {
+                token.filterField = 'MailingCity__c';
+                token.filterValue = 'Brooklyn';
+            }
+        }
+        
+        TDTM_Global_API.setTdtmConfig(tokens);
+        
+        //Update address records for a positive and negative outcome 
+        for (Integer i = 0; i < addresses.size(); i++) {
+            Address__c addr = addresses[i];
+            if (i == 0) {
+                addr.Default_Address__c = FALSE; 
+            } else if (i == 1) {
+                addr.MailingCity__c = 'Brooklyn';
+                addr.Default_Address__c = FALSE; 
+            } 
+        }
+        
+        update addresses; 
+        
+        List<Address__c> returnAddresses = [SELECT Id, MailingCity__c, Latest_Start_Date__c, Latest_End_Date__c, Default_Address__c
+                                            FROM Address__c 
+                                            WHERE Id IN :addresses]; 
```

Comment:
I didn't even realize we could do this if I am honest...we don't have to say in :addressIds or some other list? (I've never even tried if I am honest...)

Reply:
@lifewithryan , sorry are you asking if we can create an actual list with just the addressIds that we can use in the query instead of the list of addresses?

Reply:
@kdy618 this was just a comment, no action needed.

Reply:
I know this has been resolved...I didn't know this worked either but it looks like it works. However, being a little picky, is passing a list of objects less efficiency than passing a list/set of ids? @kdy618 @lifewithryan

Reply:
@xinnanlisf , I'm open for the option of creating a new list of just Ids. I think the reason why I did not initially is because I assumed creating another list of Ids would be expensive if it's just used to it for the soql query and not used elsewhere.